src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -525,7 +532,7 @@ def mgr_map_has_standby(self) -> bool:
 class MdsService(CephService):
     TYPE = 'mds'
 
-    def config(self, spec: ServiceSpec) -> None:
+    def config(self, spec: ServiceSpec, daemon_id: str) -> None:
```

Comment:
#type: ignore?

Reply:
```python
class Spec: ...

class RGWSpec(Spec): ...

class CephService:
    def config(self, spec: Spec): ...

class RGWService(CephService):
    def config(self, spec: RGWSpec): ...
```

```
file.py:9: error: Argument 1 of "config" is incompatible with supertype "CephService"; supertype defines the argument type as "Spec"
file.py:9: note: This violates the Liskov substitution principle
file.py:9: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
Found 1 error in 1 file (checked 1 source file)
```

This is only required in derived classes that demand a subclass of ServicSpec